### PR TITLE
use hardfork infrastructure in latency

### PR DESCRIPTION
**Pull request by paweljakubas**
# Issue Number

#2021 

# Overview

- [x] I have added HF infra

# Comments

How to run locally now : `stack bench cardano-wallet:latency`

At this moment I have : 
```
latency: /home/pawel/Work/IOHK/cardano-wallet/lib/shelley/test/data/cardano-node-shelley/faucet-addrs/faucet101.addr: openBinaryFile: does not exist (No such file or directory)
```
which is surprising, faucet101.addr should not be tried :thinking:

### CODE REVIEW

File: lib/shelley/bench/Latency.hs [hunk 1]
```diff
@@ -362,3 +401,19 @@ benchWithShelleyServer tracers action = do
...
+    testPoolConfigs =
+        [ -- This pool should never retire:
+          PoolConfig {retirementEpoch = Nothing}
+        -- This pool should retire almost immediately:
+        , PoolConfig {retirementEpoch = Just 3}
+        -- This pool should retire, but not within the duration of a test run:
+        , PoolConfig {retirementEpoch = Just 100_000}
+        -- This pool should retire, but not within the duration of a test run:
+        , PoolConfig {retirementEpoch = Just 1_000_000}
+        ]
```
Comment (KtorZ) [inline lib/shelley/bench/Latency.hs]: I don't think we need to have stake pools at all in the API latency benchmarks, do we?

File: lib/shelley/bench/Latency.hs [hunk 2]
```diff
@@ -345,11 +364,31 @@ benchWithShelleyServer tracers action = do
...
+                (afterFork dir)
+                (onClusterStart act dir)
+    onByron _ = pure ()
+    afterFork dir _ = do
+        let encodeAddr = T.unpack . encodeAddress @'Mainnet
+        let addresses = map (first encodeAddr) shelleyIntegrationTestFunds
+        sendFaucetFundsTo stdoutTextTracer dir addresses
+        let rewards = (,Coin $ fromIntegral oneMillionAda) <$>
+                concatMap genRewardAccounts mirMnemonics
+        moveInstantaneousRewardsTo stdoutTextTracer dir rewards
```
Comment (KtorZ) [inline lib/shelley/bench/Latency.hs]: The MIR certificates aren't useful in the latency benchmarks, only the standard faucets are needed.

File: lib/shelley/bench/Latency.hs (hunk 1 above)
Comment (paweljakubas) [reply lib/shelley/bench/Latency.hs]: indeed, fixed in https://github.com/input-output-hk/cardano-wallet/pull/2046/commits/e527debc14af47fcb558e432b16b443388364d68

File: lib/shelley/bench/Latency.hs (hunk 2 above)
Comment (paweljakubas) [reply lib/shelley/bench/Latency.hs]: indeed, fixed in https://github.com/input-output-hk/cardano-wallet/pull/2046/commits/e527debc14af47fcb558e432b16b443388364d68

**Comment by KtorZ** [comment]
Nightly build running here: https://buildkite.com/input-output-hk/cardano-wallet-nightly/builds/618

**Comment by KtorZ** [comment]
new build: https://buildkite.com/input-output-hk/cardano-wallet-nightly/builds/620

**Review by KtorZ — APPROVED** [review]

**Pull request merged**